snapd-glib/requests/snapd-json.c

```diff
@@ -101,6 +103,30 @@ _snapd_json_get_object (JsonObject *object, const gchar *name)
         return NULL;
 }
 
+const GPtrArray*
+_snapd_json_get_links (JsonObject *object, const gchar *name, GPtrArray *default_value)
+{
+    JsonObject *links_obj = json_object_get_object_member(object, "links");
+    JsonNode *node = json_node_new(JSON_NODE_OBJECT);
+    json_node_set_object(node, object);
+    gchar *object_str = json_to_string(node, TRUE);
+    g_print("JsonObject: %s\n", object_str);
+    g_free(object_str);
+    JsonArray *json_array = json_object_get_array_member(links_obj, name);
+    GPtrArray *ptr_array = g_ptr_array_new();
+    if (json_array != NULL){
```

Comment:
This is simpler as `if (json_array == NULL) { return default_value; }`, then the following code doesn't need to be indented.